src/main/java/com/actiontech/dble/server/parser/RwSplitServerParse.java

```diff
@@ -72,19 +71,19 @@ public int parse(String stmt) {
                     break;
                 case 'G':
                 case 'g':
-                    rt = gCheck(stmt, i);
+                    rt = serverParseValidations.gCheck(stmt, i);
                     break;
                 case 'I':
                 case 'i':
                     rt = iCheck(stmt, i);
                     break;
                 case 'M':
                 case 'm':
-                    rt = migrateCheck(stmt, i);
+                    rt = serverParseValidations.migrateCheck(stmt, i);
                     break;
                 case 'O':
                 case 'o':
-                    rt = optimizeCheck(stmt, i);
+                    rt = serverParseValidations.optimizeCheck(stmt, i);
                     break;
                 case 'P':
                 case 'p':
```

Comment:
whether prepareCheck is also repeated?

Reply:
Yes, I have extracted that as a common method now.